aea/agent_loop.py

```diff
@@ -259,67 +252,27 @@ def _create_tasks(self) -> List[Task]:
         :return: list of asyncio Tasks
         """
         tasks = [
-            self._task_process_inbox(),
-            self._task_process_internal_messages(),
-            self._task_process_new_skill_components(),
+            self._process_messages(HandlerItemGetter(self._message_handlers())),
+            self._task_register_periodic_tasks(),
             self._task_wait_for_error(),
         ]
         return list(map(self._loop.create_task, tasks))  # type: ignore  # some issue with map and create_task
 
-    async def _task_process_inbox(self) -> None:
-        """Process incoming messages."""
-        inbox: InBox = self._agent.inbox
-        self.logger.info("Start processing messages...")
-        while self.is_running:
-            await inbox.async_wait()
-            self._agent.react()
+    def _message_handlers(self) -> List[Tuple[Callable[[Any], None], Callable]]:
+        """Get all agent's message handlers."""
+        return self._agent.get_message_handlers()
 
-    async def _task_process_internal_messages(self) -> None:
-        """Process decision maker's internal messages."""
-        queue = self._agent.filter.decision_maker_out_queue
+    async def _process_messages(self, getter: HandlerItemGetter) -> None:
+        """Process message from ItemGetter."""
+        self.logger.info("Start processing messages...")
+        self._state.set(AgentLoopStates.started)
         while self.is_running:
-            msg = await queue.async_get()
-            # TODO: better interaction with agent's internal messages
-            self._agent.filter._process_internal_message(  # pylint: disable=protected-access # TODO: refactoring!
-                msg
-            )
+            handler, item = await getter.get()
+            handler(item)
 
-    async def _task_process_new_skill_components(self) -> None:
+    async def _task_register_periodic_tasks(self) -> None:
         """Process new behaviours added to skills in runtime."""
         while self.is_running:
             # TODO: better handling internal messages for skills internal updates
-            self._agent.filter._handle_new_behaviours()  # pylint: disable=protected-access # TODO: refactoring!
-            self._agent.filter._handle_new_handlers()  # pylint: disable=protected-access # TODO: refactoring!
             self._register_periodic_tasks()  # re register, cause new may appear
             await asyncio.sleep(self.NEW_BEHAVIOURS_PROCESS_SLEEP)
-
-
-class SyncAgentLoop(BaseAgentLoop):
```

Comment:
Didn't we say we'd keep it and just impose an ordering on events?

Reply:
will be re implemented after async version completed.

Reply:
```
SyncAgentLoop = AsyncAgentLoop
# Temporary stub
```

Reply:
done